hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/OnJoinOp.java

```diff
@@ -86,6 +90,18 @@ public void run() throws Exception {
                     getLogger().warning("Error while running post-join operation: " + op, e);
                 }
             }
+
```

Comment:
Just wondering: does this affect rolling upgrade in any way? If we have members of mixed versions, can all members still join the cluster?

Reply:
Good question, since this is for 4.0, this is fine but yes, this fix is not compatible with patch or minor releases. Since we need to backport this to a patch release, maybe we can do something like keep the old behaviour but add some additional place where we broadcast the join operations.

Reply:
Ah, I overlooked the fact that this is `master`. I was indeed thinking about the 3.12 patch release.